server/src/infra/ldap_handler.rs

```diff
@@ -120,15 +121,51 @@ fn get_user_attribute(user: &User, attribute: &str, dn: &str) -> Result<Option<V
     }))
 }
 
+fn expand_attribute_wildcards(attributes: &[String], all_attribute_keys: &[&str]) -> Vec<String> {
+    let mut attributes_out = attributes.to_owned();
+
+    if attributes_out.iter().any(|x| x == "*") || attributes_out.is_empty() {
+        debug!(r#"Expanding *"#);
+        // Splice in all non-operational attributes
+        attributes_out.extend(all_attribute_keys.iter().map(|s| s.to_string()));
+        // Remove occurrences of '*'
+        attributes_out.retain(|x| x != "*");
+    }
+
+    if attributes_out.iter().any(|x| x == "+") {
+        debug!(r#"Removing +"#);
+        attributes_out.retain(|x| x != "+");
```

Comment:
Does the same go for `*`, or since we intend to handle `*` by first expanding & removing it, should we actually throw an error in `get_user_attribute` if somehow a `*` makes it that far?

I'll rephrase:

Should we
1. remove `*` in expansion, but retain `+`, 
   - ignore `+` in resolution
   - throw an error when resolving `*`, but give a useful error message
2. retain both in expansion, ignore both in resolution

Reply:
Yes, 1 is better. Actually, in the attribute translation we should ignore rather than error on unknown attributes: it was useful in the beginning, but now we have enough support to get away with ignoring, I think. Maybe we can still log ignored attributes with debug!